### PR TITLE
Sync 4.2.1

### PR DESCRIPTION
**Pull request by uilianries**

### CODE REVIEW

**Review by ericLemanissier — APPROVED** [review]

File: conanfile.py [hunk 1]
```diff
@@ -198,6 +198,8 @@ def system_requirements(self):
                 package_tool.install(package)
 
     def _copy_pkg_config(self, name):
+        if not name in self.deps_cpp_info.deps:
```
Comment (Croydon) [inline conanfile.py]: Could you please explain this further?

Comment (Croydon) [reply conanfile.py]: @ericLemanissier ^

Comment (ericLemanissier) [reply conanfile.py]: It could be a workaround for a bug in conan 1.28.0. I never tested without this change on conan 1.28.1. It's harmless anyway

**Comment by Croydon** [comment]
> ERROR: vorbisenc not found using pkg-config

**Comment by ericLemanissier** [comment]
Sorry, I'm mobile only (during vacation), so I cannot debug this currently

**Comment by ericLemanissier** [comment]
It could be fixed by https://github.com/conan-io/conan/pull/7524
I will retry this pr with the next release of conan

**Comment by ericLemanissier** [comment]
I fixed the vorbisenc thing, but it now fails with :
```
pkg-config --exists --print-errors xcb >= 1.4
Requested 'xcb >= 1.4' but version of xorg-xcb is system
```
So this should be solved by https://github.com/conan-io/conan-center-index/pull/2741

**Comment by ericLemanissier** [comment]
I think we're good to go !

**Pull request merged**